crates/ra_assists/src/assists/flip_trait_bound.rs

```diff
@@ -0,0 +1,184 @@
+//! `flip_trait_bound` provides an assist that flips the positions of adjacent
+//! type bounds in a type bound list.
+
+use crate::{Assist, AssistCtx, AssistId};
+use hir::db::HirDatabase;
+use ra_syntax::ast::{AstNode, TypeBound, TypeBoundList};
+use ra_syntax::{NodeOrToken, TextRange};
+use std::cmp::{max, min};
+
+/// Flip trait bound assist.
+pub(crate) fn flip_trait_bound(mut ctx: AssistCtx<impl HirDatabase>) -> Option<Assist> {
+    // The assist should only be applied when the cursor is in a TypeBoundList
+    let bound_list = ctx.node_at_offset::<TypeBoundList>()?;
```

Comment:
Yeah, I think we should focus on `+` here, and check that it's parent is `TypeBoundList`